src/univariate/affine.jl

```diff
@@ -0,0 +1,241 @@
+parameters `μ`, `σ`, `base_dist` is the distribution of the random variable
+```math
+X = μ + σ * Z
+```
+
+If `base_dist` is a discrete distribution, the probability mass function of the transformed 
+distribution is given by
+```math
+P(X = x) = P\\left(Z = \\frac{x-μ}{σ} \\right).
+```
+
+If `base_dist` is a continuous distribution, the probability density function of
+the transformed distribution is given by
+```math
+f(x) = \\frac{1}{σ} base_dist \\! \\left( \\frac{x-μ}{σ} \\right).
+```
+
+If `base_dist` falls under a location-scale family, `affine` will attempt to return a 
+transformed distribution of type `base_dist` if an `affine` method has been implemented. 
+Otherwise, it will fall back on the `AffineDistribution` wrapper type.
+
+```julia
+d = σ * (base_dist + μ)     # Create location-scale transformed distribution
+params(d)                   # Get the parameters, i.e. (μ, scale, base_dist)
+location(d)                 # Get the location parameter
+scale(d)                    # Get the scale parameter
+```
+"""
+struct AffineDistribution{
+        F<:VariateForm,
+        S<:ValueSupport,
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike,
+        D<:Distribution{F,S}
+    } <: Distribution{F,S}
+    μ::Tμ
+    σ::Tσ
+    base_dist::D
+
+    function AffineDistribution(
+        μ::Tμ, σ::Tσ, base_dist::D; 
+        check_args=true
+        ) where {
+            F<:VariateForm,
+            S<:ValueSupport,
+            D<:Distribution{F, S},
+            Tμ<:ArrayLike,
+            Tσ<:ArrayLike
+        }
+        !check_args || @check_args(AffineDistribution, !iszero(σ))
+        return new{F,S,Tμ,Tσ,D}(μ, σ, base_dist)
+    end
+end
+
+
+# Constructors
+
+"""
+    +(shift, base_dist)
+
+Return a version of `base_dist` that has been translated by `shift`, where `shift` can be a
+real number or (for a multivariate distribution) a vector.
+"""
+function Base.:+(μ::ArrayLike, base_dist::Distribution)
+    return AffineDistribution(μ, mult_identity(μ), base_dist)
+end
+
+"""
+    *(scale_factor, base_dist)
+
+Return a version of `base_dist` that has been scaled by `scale`, where `scale` can be a real
+number or (for a multivariate distribution) matrix.
+"""
+function Base.:*(σ::ArrayLike, base_dist::Distribution)
+    return AffineDistribution(add_identity(σ), σ, base_dist)
+end
+
+
+function Base.:+(μ::ArrayLike, aff_dist::AffineDistribution)
+    return AffineDistribution(μ + aff_dist.μ, aff_dist.σ, aff_dist.base_dist)
+end
+
+function Base.:*(σ::ArrayLike, aff_dist::AffineDistribution)
+    return AffineDistribution(σ * aff_dist.μ, σ * aff_dist.σ, aff_dist.base_dist)
+end
+
+
+Base.:+(d::Distribution, μ::ArrayLike) = μ + d
+Base.:-(d::UnivariateDistribution) = -1 * d
+Base.:-(d::Distribution, μ::ArrayLike) = d + -μ
+Base.:-(μ::ArrayLike, d::Distribution) = μ + -d
+Base.:*(d::Distribution, σ::ArrayLike) = σ * d
+Base.:/(d::Distribution, τ::ArrayLike) = inv(τ) * d
```

Comment:
I see the problems with the other methods being too general, but shouldn't these be fine if I define the other methods more specifically (since these are just syntactic sugar reducing to the other methods)?

Reply:
Oh, perhaps `/` and `*` are too general, since matrix multiplication isn't commutative.

Reply:
Generally, `Distribution` is too general.